python/lsst/obs/base/ingest.py

```diff
@@ -220,6 +232,12 @@ def extractMetadata(self, filename: str) -> RawFileData:
             as well as the original filename.  All fields will be populated,
             but the `RawFileData.dataId` attribute will be a minimal
             (unexpanded) `DataCoordinate` instance.
+
+        Notes
+        -----
+        Assumes that there is a single dataset associated with the given
+        file.  This method must be subclassed if multiple datasets are
+        stored in a single raw file.
```

Comment:
I suspect we'll need to refactor this as well for DECam ingest, because it would want to delegate to `super()` for a lot of the logic below, and probably can't in its current form.  But I guess that's best left for @parejkoj when he actually does that work, since otherwise it'll be hard to know if we refactored it correctly?

Reply:
I did remove the big chunk of region code out of the method.

Reply:
Let me see if I can refactor it a bit more.

Reply:
I've pushed a refactor. I won't merge until @parejkoj comments. Now I don't think people have to call super if they refactor extractMetadata. They can loop over HDUs, read new headers and pass each header to the new refactored routine.

Reply:
"This method must be subclassed" - do you mean "subclasses must override this method ..."?

Reply:
Yes. You must subclass the method. What's wrong with what I said?

Reply:
The phrase "subclass the method" strikes me as odd. You're not subclassing the method, you're subclassing the class and overriding the method. Is the former a term of art I've missed?

Reply:
It's a phrasing I've used for 20 years. Anyhow, I've changed the wording.